File: packages/dart/lib/src/distance/distance.dart
import 'dart:math' as math;

const double _earthRadiusKm = 6371.0088;

double distance(List<double> a, List<double> b) {
  final lng1 = a[0];
  final lat1 = a[1];
  final lng2 = b[0];
  final lat2 = b[1];

  final dLat = (lat2 - lat1) * math.pi / 180;
  final dLng = (lng2 - lng1) * math.pi / 180;

  final phi1 = lat1 * math.pi / 180;
  final phi2 = lat2 * math.pi / 180;

  final h = math.sin(dLat / 2) * math.sin(dLat / 2) +
      math.cos(phi1) * math.cos(phi2) * math.sin(dLng / 2) * math.sin(dLng / 2);

  return 2 * _earthRadiusKm * math.asin(math.sqrt(h));
}

File: packages/rust/src/distance/distance.rs
const EARTH_RADIUS_KM: f64 = 6371.0088;

pub fn distance(a: [f64; 2], b: [f64; 2]) -> f64 {
    let lat1 = a[1].to_radians();
    let lat2 = b[1].to_radians();
    let dlat = (b[1] - a[1]).to_radians();
    let dlng = (b[0] - a[0]).to_radians();

    let h = (dlat / 2.0).sin().powi(2)
        + lat1.cos() * lat2.cos() * (dlng / 2.0).sin().powi(2);

    2.0 * EARTH_RADIUS_KM * h.sqrt().asin()
}
